app/qml/ExternalResourceBundle.qml

```diff
@@ -39,17 +40,35 @@ Item {
             }
         }
 
-        property var imageSelected: function imageSelected(imagePath) {
-            var homePath  = featureForm.project ? featureForm.project.homePath : ""
-            var fileName = QgsQuick.Utils.getRelativePath(imagePath, homePath)
-            if (!fileName) {
-                fileName = __inputUtils.getFileName(imagePath)
-            }
-            if (!QgsQuick.Utils.fileExists(homePath, fileName)) {
-                __inputUtils.copyFile(imagePath, homePath + "/" + fileName)
-            }
-            externalResourceHandler.itemWidget.valueChanged(fileName, false)
+        property var confirmImage: function confirmImage(itemWidget, prefixToRelativePath, value) {
+          var newPath = __inputUtils.renameFile(prefixToRelativePath + "/" + value, customPrefix)
+          if (newPath) {
+            externalResourceHandler.itemWidget = itemWidget
```

Comment:
do we need to do the association of itemWidget, then some calls and then set itemWidget to undefined again?

Reply:
Removing reference is just kind of good will to avoid possible conflicts in case of multiple external source fields. However, this shouldn't happen at all since reference is set to bundle every time when value is going to be changed. Therefore this "cleanup" can be removed.
 
There is even a case, when a user choose image, cancel a filedialog and the whole workflow is at its end, but reference hasnt been removed.

Reply:
I am mainly interested why we need to temporarily assign the widget to to the handler at all... instead of just doing something like:
```
itemWidget.image.currentValue = newCurrentValue
itemWidget.valueChanged(...)
```
If it is necessary to assign it, then it's ok, but it would be good to have a comment why - and the reset to 'undefined' is a good idea.

Reply:
For most actions, there is no need for that. However, while choosing an image from a gallery, there are callbacks either on ok and cancel, when a externalHandler has to have a reference to a correct field's widget to finish a procedure. Just wanted to have all the same at the beginning. 

Going to remove assigning a widget to the handler if not needed.